wagtail/models/__init__.py

```diff
@@ -217,7 +217,171 @@ class Meta:
         abstract = True
 
 
-class Page(AbstractPage, index.Indexed, ClusterableModel, metaclass=PageBase):
+class RevisionMixin:
+    """A mixin that allows a model to have revisions."""
+
+    @property
+    def base_content_type(self):
+        # Set default value for base_content_type to the content_type.
+        # This distinction is useful for models that use inheritance.
+        return self.get_content_type()
+
+    @property
+    def revisions(self):
+        # This acts as a replacement for Django's related manager since we don't
+        # use a GenericRelation/GenericForeignKey.
+        return Revision.objects.filter(
+            base_content_type=self.base_content_type,
+            object_id=self.pk,
+        )
+
+    @property
+    def approved_schedule(self):
+        return self.revisions.exclude(approved_go_live_at__isnull=True).exists()
+
+    def get_content_type(self):
+        if hasattr(self, "content_type"):
+            return self.content_type
+        return ContentType.objects.get_for_model(self)
```

Comment:
Thoughts? Can't make this a `content_type` `@property` as the model (e.g. `Page`) might already have it as a field, so I wrote this as a `get_content_type()` method instead.